src/Microsoft.Extensions.Primitives/StringTokenizer.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Primitives
+{
+    /// <summary>
+    /// Tokenizes a <c>string</c> into <see cref="StringSegment"/>s.
+    /// </summary>
+    public struct StringTokenizer :  IEnumerable<StringSegment>
+    {
+        private readonly string _value;
+        private readonly char _separator;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="StringTokenizer"/>.
+        /// </summary>
+        /// <param name="value">The <c>string</c> to tokenize.</param>
+        /// <param name="separator">The character to tokenize by.</param>
+        public StringTokenizer(string value, char separator)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            _value = value;
+            _separator = separator;
+        }
+
+        /// <inheritdoc />
+        public IEnumerator<StringSegment> GetEnumerator()
+        {
+            return new Enumerator(ref this);
```

Comment:
I was following the pattern from https://github.com/aspnet/Common/blob/dev/src/Microsoft.Extensions.Primitives/StringValues.cs#L209. Although it might not be super relevant to how GetEnumerator gets used in this context.